api/lds.proto

```diff
@@ -66,13 +66,27 @@ message FilterChainMatch {
 
   // If non-empty, an IP address and prefix length to match addresses when the
   // listener is bound to 0.0.0.0/::.
```

Comment:
I think it's simpler to understand. My only question is what happens if we decide to pick filter chain A from IP Y's FilterChainMatch table and there is also a Listener for IP X with filter chain B? Do we expect the user to ensure A and B are consistent? Or do we expect that we won't have a listener for IP X when using `use_original_dst` this way?